addons/board/static/src/board_controller.js

```diff
@@ -102,9 +102,16 @@ export class BoardController extends Component {
     }
 
     saveBoard() {
+        const templateFn = renderToString.app.getTemplate("board.arch");
+        const bdom = templateFn(this.board, {});
+        const root = document.createElement("rendertostring");
+        blockDom.mount(bdom, root);
+        const result = xmlSerializer.serializeToString(root);
+        const arch = result.slice(result.indexOf('<', 1), result.indexOf('</rendertostring>'));
+
         this.rpc("/web/view/edit_custom", {
             custom_id: this.board.customViewId,
-            arch: renderToString("board.arch", this.board),
+            arch,
```

Comment:
I tested, and I don't see any difference in the output arch between your version (with the xml serializer) and the previous one simply using renderToString. What am I missing?

Reply:
Hello @aab-odoo,
I have tested again on a runbot 16.0 and [a runbot build of this PR](https://runbot.odoo.com/runbot/batch/1033275), and I could reproduce the issue.
In order to reproduce it, here are the steps that I follow:
- Go to any list view (e.g. Apps)
- Add a custom filter on a date, such as Last Updated on is after "05/26/2023 00:00:00" (so that there is a < or > in the domain)
- Favorites > Add to my dashboard
- Go to the dashboard and move the panel to some other position (this will actually call saveBoard)
- Go to the debug=1 menu Customized Views and open the modified view

On the runbot 16.0, I get:
```xml
<form string="My Dashboard"><board style="2-1"><column></column><column><action name="36" string="Apps" view_mode="kanban" context="{'orderedBy': [], 'group_by': [], 'dashboard_merge_domains_contexts': False}" domain="[&amp;#x27;&amp;amp;&amp;#x27;, &amp;#x27;!&amp;#x27;, [&amp;#x27;name&amp;#x27;, &amp;#x27;=like&amp;#x27;, &amp;#x27;theme_%&amp;#x27;], [&amp;#x27;write_date&amp;#x27;, &amp;#x27;&amp;gt;&amp;#x27;, &amp;#x27;2023-05-25 22:00:00&amp;#x27;]]" fold="0"></action></column><column></column></board></form>
```
whereas on my PR's runbot I get:
```xml
<form string="My Dashboard"><board style="2-1"><column></column><column><action name="36" string="Apps" view_mode="kanban" context="{'orderedBy': [], 'group_by': [], 'dashboard_merge_domains_contexts': False}" domain="[['write_date', '&gt;', '2023-05-25 22:00:00']]" fold="0"></action></column><column></column></board></form>
```

Note that all the single quotes have been escaped twice, giving `&amp;#x27;` instead of `&#x27;` which would have been enough. The migration scripts will decode it once, and have a domain with these escaped quotes, and they won't be able to interpret it as a serialized python domain.

Reply:
The upgrade scripts have been adapted[^1] to do the double escaping to fix those domains

[^1]: https://github.com/odoo/upgrade/commit/4cad53cd31048f545463c5c339c0b10fc7ca350d

Reply:
> Hello @aab-odoo, I have tested again on a runbot 16.0 and [a runbot build of this PR](https://runbot.odoo.com/runbot/batch/1033275), and I could reproduce the issue. In order to reproduce it, here are the steps that I follow:
> 
> * Go to any list view (e.g. Apps)
> * Add a custom filter on a date, such as Last Updated on is after "05/26/2023 00:00:00" (so that there is a `<` or `>` in the domain)
> * Favorites > Add to my dashboard
> * Go to the dashboard and move the panel to some other position (this will actually call saveBoard)
> * Go to the debug=1 menu Customized Views and open the modified view
> 
> On the runbot 16.0, I get:
> 
> ```
> <form string="My Dashboard"><board style="2-1"><column></column><column><action name="36" string="Apps" view_mode="kanban" context="{'orderedBy': [], 'group_by': [], 'dashboard_merge_domains_contexts': False}" domain="[&amp;#x27;&amp;amp;&amp;#x27;, &amp;#x27;!&amp;#x27;, [&amp;#x27;name&amp;#x27;, &amp;#x27;=like&amp;#x27;, &amp;#x27;theme_%&amp;#x27;], [&amp;#x27;write_date&amp;#x27;, &amp;#x27;&amp;gt;&amp;#x27;, &amp;#x27;2023-05-25 22:00:00&amp;#x27;]]" fold="0"></action></column><column></column></board></form>
> ```
> 
> whereas on my PR's runbot I get:
> 
> ```
> <form string="My Dashboard"><board style="2-1"><column></column><column><action name="36" string="Apps" view_mode="kanban" context="{'orderedBy': [], 'group_by': [], 'dashboard_merge_domains_contexts': False}" domain="[['write_date', '&gt;', '2023-05-25 22:00:00']]" fold="0"></action></column><column></column></board></form>
> ```
> 
> Note that all the single quotes have been escaped twice, giving `&amp;#x27;` instead of `&#x27;` which would have been enough. The migration scripts will decode it once, and have a domain with these escaped quotes, and they won't be able to interpret it as a serialized python domain.

Ok now I see. I don't know what I did/checked earlier. Thanks